src/main/java/com/google/devtools/build/lib/rules/repository/NewLocalRepositoryFunction.java

```diff
@@ -186,22 +188,10 @@ private static ResolvedEvent resolve(Rule rule, BlazeDirectories directories) {
             .append(", path = ")
             .append(Starlark.repr(pathObj));
 
-    Object buildFileObj = rule.getAttr("build_file");
-    if ((buildFileObj instanceof String) && !((String) buildFileObj).isEmpty()) {
-      // Build files might refer to an embedded file (as they to for "local_jdk"), so we have to
-      // describe the argument in a portable way.
-      origAttr.put("build_file", buildFileObj);
-      String buildFileArg;
-      PathFragment pathFragment = PathFragment.create((String) buildFileObj);
-      PathFragment embeddedDir = directories.getEmbeddedBinariesRoot().asFragment();
-      if (pathFragment.isAbsolute() && pathFragment.startsWith(embeddedDir)) {
-        buildFileArg =
-            "__embedded_dir__ + \"/\" + "
-                + Starlark.repr(pathFragment.relativeTo(embeddedDir).toString());
-      } else {
-        buildFileArg = Starlark.repr(buildFileObj.toString());
-      }
-      repr.append(", build_file = ").append(buildFileArg);
+    Label buildFile = (Label) rule.getAttr("build_file", BuildType.NODEP_LABEL);
```

Comment:
Why not `WorkspaceAttributeMapper.of(rule).get("build_file", Type.NODEP_LABEL) `?

That seems to be the canonical way to access attributes in the (few) native implementations of workspace rules.

Reply:
At least you get a more reasonable exception than `ClassCastException` if you do this. I won't insist, though, it's not a big difference.

Reply:
thing is, you won't get an exception at all; it's impossible for a label-typed attribute to have a non-label value.